malariagen_data/anoph/snp_frq.py

```diff
@@ -1252,6 +1255,398 @@ def snp_genotype_allele_counts(
+        values less than 70, green for values greater
+        than 90, and a gradient in between for the
+        values in between.
+        """
+        if isinstance(val, str):
+            if re.search("NON_SYNONYMOUS", val):
+                return "background-color: red"
+            elif re.search("SYNONYMOUS", val):
+                return "background-color: yellow"
+            else:
+                return ""
+        elif isinstance(val, float):
+            if np.isnan(val):
+                return ""
+            else:
+                r = int(255 * (1 - val))
+                g = 255
+                b = 255
+                return f"background-color: rgb({r},{g},{b})"
+        else:
+            return ""
+
+    def _move_transcript_to_column_names(self, t1, impact_columns):
+        t1_transcript = np.unique(t1["transcript"])[0]
+        t1 = t1.rename(
+            columns={x: f"{x}_{t1_transcript}" for x in impact_columns}
+        ).drop(columns="transcript")
+        return t1
+
+    # Function to merge two transcript tables
+    def _merge_transcript_tables(self, t1, t2, impact_columns):
+        if "transcript" in t1.columns:
+            t1 = self._move_transcript_to_column_names(t1, impact_columns)
+        if "transcript" in t2.columns:
+            t2 = self._move_transcript_to_column_names(t2, impact_columns)
+
+        # FIXME: this needs to be made agnostic to species groups, e.g. funestus.
+        merged_df = pd.merge(
+            t1,
+            t2,
+            on=[
+                "contig",
+                "position",
+                "ref_allele",
+                "alt_allele",
+                "pass_gamb_colu_arab",
+                "pass_gamb_colu",
+                "pass_arab",
+            ],
+            how="outer",
+        )
+        return merged_df
+
+    def _any_snp_allele_freq(
+        self,
+        region,
+        array_of_snps,
+        cohorts=None,
+        min_cohort_size=1,
+        sample_query=None,
+        sample_query_options=None,
+        site_mask=None,
+        sample_sets=None,
+        drop_invariant=False,
+        include_counts=False,
+    ):
+        # Access sample metadata.
+        df_samples = self.sample_metadata(
+            sample_sets=sample_sets,
+            sample_query=sample_query,
+            sample_query_options=sample_query_options,
+        )
+        num_samples = df_samples.shape[0]
+
+        # Build cohort dictionary, maps cohort labels to boolean indexers.
+        if cohorts is None:
+            # Create cohorts full all samples, then each species, then each country x species combo
+            # FIXME: this needs to be made agnostic to species groups, e.g. funestus.
+            coh_dict1 = {
+                "all": np.full(num_samples, True),
+                "gambiae": df_samples.taxon == "gambiae",
+                "coluzzii": df_samples.taxon == "coluzzii",
+                "arabiensis": df_samples.taxon == "arabiensis",
+            }
```

Comment:
```suggestion
            coh_dict1 = {t : df_samples.taxon == t for t in np.unique(df_samples.taxon)}
            coh_dict1['all'] = np.full(num_samples, True)
```

Reply:
this can easily be generalised using the metadata

Reply:
Yes. Though I think it would still be nice to include the "all" cohort.

Reply:
> Yes. Though I think it would still be nice to include the "all" cohort.

its there :)